openedx/openedx-platform
# feat: [FC-0047] add mobile push notifications functionality

## Description
This PR addressed to adding the ability to send push notifications to the OeX mobile app on IOS and Android, using edx-ace functionality and Firebase Cloud Messaging as a cloud provider.

Some push notifications are also added here directly, namely: 
  - push notifications for user enroll;
  - push notifications for user unenroll;
  - push notifications for add course beta testers;
  - push notifications for remove course beta testers;
  - push notification event to discussions.

Useful information to include:

- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author",
"Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these
changes.

## Supporting information

FC-0047
TBD

## Testing instructions

1. Install edx-ace version 1.9.0.
2. Add FIREBASE_CREDENTIALS or FIREBASE_CREDENTIALS_PATH to the platform settings.
3. Add a new `push_notification` channel to ACE_ENABLED_CHANEL.
4. Log in to OeX.
5. Install the OeX mobile application on your device.
6. Add your device token to FCM Devices with a link to your user from the admin panel.
7. Enroll/unenroll the user(from step 7) in the “Instructor” tab to trigger the push notification event.
8. Make sure you receive push notifications on your device.

## Deadline
"None"

## Other information
 - Edx-ace push notifications ARD - https://github.com/openedx/edx-ace/pull/277
 - This PR depends on Edx-ace [PR](https://github.com/openedx/edx-ace/pull/282) which adds the necessary functionality.
